solidity/contracts/system/KeepFactorySelection.sol

```diff
@@ -139,40 +139,55 @@ library KeepFactorySelection {
         );
         _self.selectedFactory = _self.factorySelector.selectFactory(
             seed,
-            _self.keepStakeFactory,
-            _self.ethStakeFactory
+            _self.keepStakedFactory,
+            _self.fullyBackedFactory
         );
 
         require(
-            _self.selectedFactory == _self.keepStakeFactory ||
-                _self.selectedFactory == _self.ethStakeFactory,
+            _self.selectedFactory == _self.keepStakedFactory ||
+                _self.selectedFactory == _self.fullyBackedFactory,
             "Factory selector returned unknown factory"
         );
     }
 
-    /// @notice Sets the address of the fully backed, ETH-stake based keep
+    /// @notice Sets the address of the KEEP-stake based keep factory. This is
+    /// a default factory used when a ETH-bond-only based keep factory is not set.
+    /// @dev Can be called multiple times! It's responsibility of a contract
+    /// using this library to limit and protect updates.
+    /// @param _keepStakedFactory Address of the regular, KEEP-stake based keep
+    /// factory.
+    function setKeepStakedKeepFactory(
```

Comment:
I don't like this name... Since we're in `KeepFactorySelection` maybe it could be named `setKeepStakedFactory` and `setFullyBackedKeepFactory` could be renamed to `setFullyBackedFactory`? 

We could even go further and rename `setKeepFactorySelector` to `setFactorySelector`...

We would use it in `TBTCSystem` as following:
```solidity
keepFactorySelection.setKeepStakedFactory(newKeepStakedFactory);
keepFactorySelection.setFullyBackedFactory(newFullyBackedFactory);
keepFactorySelection.setFactorySelector(newFactorySelector);
```

Thoughts?

Reply:
Per your question earlier on flowdock, should we just combine both factory settings into a single `setFactories` call?

Reply:
We have it combined on `TBTCSystem` level, we can do it also further on the `KeepFactorySelection` library level.

Reply:
I agree with @Shadowfiend and I'd combine it. It's simpler.

Reply:
c09fb403